verilog/analysis/verilog_project.h

```diff
@@ -165,6 +165,30 @@ class InMemoryVerilogSourceFile : public VerilogSourceFile {
   const absl::string_view contents_for_open_;
 };
 
+// Source file that was already parsed and got its own TextStructure.
+// Doesn't require file-system access, nor create temporary files.
+class ParsedVerilogSourceFile : public VerilogSourceFile {
+ public:
+  // filename can be fake, it is not used to open any file.
+  ParsedVerilogSourceFile(absl::string_view filename,
+                          const verible::TextStructureView& text_structure,
+                          absl::string_view corpus = "")
+      : VerilogSourceFile(filename, filename, corpus),
+        text_structure_(text_structure) {}
+
+  // Do nothing (file contents already loaded)
+  absl::Status Open() override;
```

Comment:
Given that we don't expect to derive further, we can make the `override` a `final` (same below; same in `InMemoryVerilogSourceFile`). Might need a `class ParsedVerilogSourceFile final : ...` above.

Reply:
I made the whole class final (same with InMemoryVerilogSourceFile class).

Reply:
Nice. At that point, the compiler knows that once it can proof that there can only be a ParsedVerilogSourceFile in a particular instance, it can replace calls to the virtual methods with direkt calls and does not have to go through the virtual method table. While here the performance win is probably not noticeable, it is a win for the reader of the code.

So for readability, I'd then also suggest to replace `override` with `final` in all the methods so that it is easier to see for the reader at first glance without having looked at the class declaration above.

Not critical for this review, can be changed later.